ocf-servers/start-ocf-servers-in-docker.sh

```diff
@@ -1,17 +1,75 @@
 #!/bin/bash
 
+# project home directory
+PROJ_HOME=/opt/SmartHome-Demo
+
+# direcotry for ocf servers
+OCF_DIR=$PROJ_HOME/ocf-servers/js-servers
+
+# configuration file name
+CONFIG_FILE=$PROJ_HOME/ocf-servers/ocf-server.conf
+
 # Print out container IP address
 echo "This container IP address is: `hostname -i`"
 
 # Set-up the path to where the node.js modules were installed
-export NODE_PATH=/opt/SmartHome-Demo/ocf-servers/node_modules/
+export NODE_PATH=$PORJ_HOME/node_modules/
+
 
+parse_error=0
 # Start all different OCF servers available
-for file in `ls -1 /opt/SmartHome-Demo/ocf-servers/js-servers/*.js`
-do
-	/usr/bin/node $file -s &
-	sleep 0.2
-done
+if [ -f "$CONFIG_FILE" ];
+then
+    while read ocf_server_name num; do
+        # skip commented lines
+        if [[ $ocf_server_name =~ "^\#.*" ]];
+        then
+            continue
+        fi
+
+        if [ -f "$OCF_DIR/$ocf_server_name.js" ];
+        then
+            if [[ $num =~ ^[0-9]+$ ]];
+            then
+                for i in $(seq 1 $num);
+                do
+                    echo "...Start $ocf_server_name server..."
+                    if [ "$i" == "1" ];
+                    then 
+                        /usr/bin/node "$OCF_DIR/$ocf_server_name.js" -s &
+                    else
+                        # create a new ocf server file
+                        \cp -fR $OCF_DIR/$ocf_server_name.js $OCF_DIR/$ocf_server_name$i.js
```

Comment:
If I'm reading this correctly, we could get in some bizarre scenario whereby we hit a parse error half-way through the processing of this file which then triggers *all* OCF server files to be started with potentially some of them having already been copied and renamed. Is this correct?

If true, this can lead to awkward behaviours but I don't think it'd be a real problem.